From b0c0409226bb05788179c52f865e0a421ffb50ae Mon Sep 17 00:00:00 2001
From: "Pillai, Venkatesh" <venkatesh.pillai@intel.com>
Date: Mon, 4 Sep 2023 14:01:44 +0530
Subject: [PATCH] Fix for Camera Issue

- Camera App crash while switching
- Camera App crash while Closing
- Camera Video Record Failed
- Multicamera Video Record Failed

Change-Id: I7544fae2478f01f45d25c4b347f4e06b2d9622af
Tracked-On: OAM-111858
Signed-off-by: Pillai, Venkatesh <venkatesh.pillai@intel.com>
---
 camera/device/default/ExternalCameraDeviceSession.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/camera/device/default/ExternalCameraDeviceSession.cpp b/camera/device/default/ExternalCameraDeviceSession.cpp
index c9629744c..c5260608e 100644
--- a/camera/device/default/ExternalCameraDeviceSession.cpp
+++ b/camera/device/default/ExternalCameraDeviceSession.cpp
@@ -233,6 +233,11 @@ void ExternalCameraDeviceSession::closeOutputThreadImpl() {
         mOutputThread->requestExitAndWait();
         mOutputThread.reset();
     }
+    if (mBufferRequestThread != nullptr) {
+        mBufferRequestThread->requestExitAndWait();
+        mBufferRequestThread.reset();
+    }
+
 }
 
 Status ExternalCameraDeviceSession::initStatus() const {
-- 
2.17.1

